Gems/Atom/RHI/Code/Source/RHI/RayTracingAccelerationStructure.cpp

```diff
@@ -319,9 +319,12 @@ namespace AZ::RHI
                 return ResultCode::Success;
             });
 
-        if (const auto& name = m_tlasBuffer->GetName(); !name.IsEmpty())
+        if (m_tlasBuffer)
         {
-            m_tlasBuffer->SetName(name);
+            if (const auto& name = m_tlasBuffer->GetName(); !name.IsEmpty())
+            {
+                m_tlasBuffer->SetName(name);
+            }
```

Comment:
Would it make sense to move this block into the lambda above instead?

Reply:
Since that is called once for the `MultiDeviceObject` and sets the name for all `DeviceObject`s, and the lambda progressively adds the device-specific `TLASBuffer`s, calling it in there either would mean doing what `SetName` is doing by hand for each device-specific `TLASBuffer` or always running through the loop, setting some names multiple times.
I'd prefer to keep it here and set it only once.